lib/server.js

```diff
@@ -28,6 +30,16 @@ function createServer (options, httpHandler) {
   } else {
     server = http.createServer(httpHandler)
     server.keepAliveTimeout = options.keepAliveTimeout
+    if (options.maxRequestsPerSocket > 0 && semver.lt(versions.node, '16.10.0')) {
+      // when node version smaller than 16.10.0 and is a number
+      // it is not a valid option
+      throw new FST_ERR_HTTP_MAX_REQUESTS_PER_SOCKET_INVALID_VERSION()
```

Comment:
This will complicate things if it is backported to previous lines - ideally it can land in v14. I would recommend not having this check and just set it.

Reply:
I will keep the `options.maxRequestsPerSocket > 0` check, as setting `0` will block all incoming connection.